cruise-control-core/src/main/java/com/linkedin/cruisecontrol/monitor/sampling/aggregator/RawMetricValues.java

```diff
@@ -202,6 +203,22 @@ public synchronized void resetWindowIndexes(long startingWindowIndex, int numWin
    * @return the aggregated values and extrapolations of the given sorted set of windows in that order.
    */
   public synchronized ValuesAndExtrapolations aggregate(SortedSet<Long> windowIndexes, MetricDef metricDef) {
+    return aggregate(windowIndexes, metricDef, true);
+  }
+
+  /**
+   * Peek the value for the current window.
+   *
+   * @param metricDef the metric definitions.
+   * @return the aggregated values and extrapolations of the given sorted set of windows in that order.
+   */
+  public synchronized ValuesAndExtrapolations peekCurrentWindow(long currentWindowIndex, MetricDef metricDef) {
+    SortedSet<Long> window = new TreeSet<>();
+    window.add(currentWindowIndex);
+    return aggregate(window, metricDef, false);
+  }
+
+  private ValuesAndExtrapolations aggregate(SortedSet<Long> windowIndexes, MetricDef metricDef, boolean checkWindow) {
```

Comment:
What is `checkWindow` used for? Can we add a comment / JavaDoc to clarify its use?